File: problems/09_heap_priority_queue/0355_design_twitter/solution.py
"""
Problem: 0355. Design Twitter
Difficulty: Medium
URL: https://leetcode.com/problems/design-twitter/

解法：每用戶 tweet list + max-heap 合併取最新 10。
"""

from __future__ import annotations

from typing import Dict, List, Set, Tuple

import heapq

class Twitter:
    def __init__(self):
        pass

    def postTweet(self, userId: int, tweetId: int) -> None:
        pass

    def getNewsFeed(self, userId: int) -> List[int]:
        pass

    def follow(self, followerId: int, followeeId: int) -> None:
        pass

    def unfollow(self, followerId: int, followeeId: int) -> None:
        pass


if __name__ == "__main__":
    twitter = Twitter()

    # Test 1
    # result = twitter.solve(...)
    # assert result == expected

    print("All tests passed!")

File: problems/09_heap_priority_queue/0355_design_twitter/solution.dart
// Problem: 0355. Design Twitter
// Difficulty: Medium

class _MaxHeap {
  // entry: [time, tweetId, uid, idx]
  final List<List<int>> a = [];
  int get size => a.length;
  void push(List<int> e) {
    // TODO: implement
  }
  List<int> pop() {
    // TODO: implement
    return [];
  }
}

class Twitter {
  int _time = 0;
  final Map<int, List<List<int>>> tweets = {}; // [time, tweetId]
  final Map<int, Set<int>> follows = {};

  void postTweet(int userId, int tweetId) {
    // TODO: implement
  }

  List<int> getNewsFeed(int userId) {
    // TODO: implement
    return [];
  }

  void follow(int followerId, int followeeId) {
    // TODO: implement
  }

  void unfollow(int followerId, int followeeId) {
    // TODO: implement
  }
}


void main() {
  // Test 1
  // final result = solve(...);
  print('All tests passed!');
}
